src/Framework/Hooks/CommonConditions.cs

```diff
@@ -112,67 +116,145 @@ private static QuestStatSummary GetQuestStats(CustomQuest managedQuest)
-
-            foreach (var ev in events)
-            {
-                flag &= Game1.player.eventsSeen.Contains(ev);
-                Monitor.VerboseLog($"Checked if event `{ev}` was seen. Current flag: {flag}");
-            }
-
-            return flag;
-        }
-
-        public static bool CheckReceivedMailCondition(string valueToCheck)
-        {
-            string[] mails = valueToCheck.Split(' ');
-            bool flag = true;
-
-            if (mails.Length < 1)
-                return false;
-
-            foreach (string mail in mails)
-            {
-                flag &= Game1.player.mailReceived.Contains(mail);
-                Monitor.VerboseLog($"Checked if mail letter `{mail}` was received. Current flag: {flag}");
-            }
-
-            return flag;
-        }
-
-        public static bool CheckFriendshipCondition(string friendshipDefinition)
-        {
-            string[] fships = friendshipDefinition.Split(' ');
-            bool flag = true;
-
-            if (fships.Length < 2)
-                return false;
-
-            for (int i = 0; i < fships.Length; i += 2)
-            {
-                string who = fships[i];
-                int currentLevel = Game1.player.getFriendshipHeartLevelForNPC(who);
-                int expectedLevel = Convert.ToInt32(fships[i + 1]);
-
-                flag &= currentLevel >= expectedLevel;
-                
-                if (Monitor.IsVerbose)
-                    Monitor.Log(
-                        $"Checked friendship level for `{who}`, " +
-                        $"current level: {currentLevel}, " +
-                        $"expected: {expectedLevel}, " +
-                        $"current flag: {flag}");
-            }
-
-            return flag;
-        }
-
+        public static bool CheckEventSeenCondition(string valueToCheck)
+        {
+            int[] events = Utility.parseStringToIntArray(valueToCheck);
+            bool flag = true;
+
+            if (events.Length < 1)
+                return false;
+
+            foreach (var ev in events)
+            {
+                flag &= Game1.player.eventsSeen.Contains(ev);
+                Monitor.VerboseLog($"Checked if event `{ev}` was seen. Current flag: {flag}");
+            }
+
+            return flag;
+        }
+
+        public static bool CheckReceivedMailCondition(string valueToCheck)
+        {
+            string[] mails = valueToCheck.Split(' ');
+            bool flag = true;
+
+            if (mails.Length < 1)
+                return false;
+
+            foreach (string mail in mails)
+            {
+                flag &= Game1.player.mailReceived.Contains(mail);
+                Monitor.VerboseLog($"Checked if mail letter `{mail}` was received. Current flag: {flag}");
+            }
+
+            return flag;
+        }
+
+        public static bool CheckFriendshipLevel(string friendshipLevel)
+        {
+            string[] flevel = friendshipLevel.Split(' ');

```

Comment:
`flevel` is not good name for variable here. I suggest `friendshipLevel` or if you preffer short names, use `fshipLevel` instead.